lib/gel/environment.rb

```diff
@@ -364,12 +364,22 @@ def self.activate(install: false, output: nil, error: true)
     @active_lockfile = true
     loader = Gel::LockLoader.new(Gel::ResolvedGemSet.load(lockfile), gemfile)
 
+    if lock_outdated?(loaded, loader)
+      write_lock(output: $stderr, lockfile: lockfile)
```

Comment:
Do we need to put the new ResolvedGemSet into `loader` after this?

Reply:
Updated! I think it would have loaded the old list of gems before. Now we read the lock file and load the dependencies in different steps.